src/Common/dataAccess/getIndexTransformationProgress.ts

```diff
@@ -0,0 +1,28 @@
+import { client } from "../CosmosClient";
+import { handleError } from "../ErrorHandlingUtils";
+import { logConsoleProgress } from "../../Utils/NotificationConsoleUtils";
+import * as Constants from "../Constants";
+import { AuthType } from "../../AuthType";
+
+export async function getIndexTransformationProgress(databaseId: string, collectionId: string): Promise<number> {
+  if (window.authType !== AuthType.AAD) {
+    return undefined;
+  }
+  let indexTransformationPercentage: number;
+  const clearMessage = logConsoleProgress(`Reading container ${collectionId}`);
+  try {
+    const response = await client()
+      .database(databaseId)
+      .container(collectionId)
+      .read({ populateQuotaInfo: true });
```

Comment:
Is this going via the SDK? The Mongo team has been hounding me about moving all stuff off of the SDK since it needs to go through legacy gateway and use the old 3.2 endpoint. It won't work for private endpoint users.

Reply:
Yes, RP doesn't support getting the progress headers. Jason told that this is the best/only way to get index progress right now. How should i be testing for private endpoints?

Reply:
This will now also be used to track index progress for SQL API